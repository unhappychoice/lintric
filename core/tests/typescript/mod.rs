use insta::assert_snapshot;
use lintric_core::analyze_code;
use serde_json;
use std::env;

#[test]
fn test_analyze_code_typescript() {
    let file_path = "tests/typescript/fixtures/basic_typescript_code.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "typescript_basic_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "typescript_basic_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_class_method_dependency() {
    let file_path = "tests/typescript/fixtures/class_method_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "class_method_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "class_method_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_import_dependency() {
    let file_path = "tests/typescript/fixtures/import_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "import_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "import_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_function_parameter_dependency() {
    let file_path = "tests/typescript/fixtures/function_parameter_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "function_parameter_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "function_parameter_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_arrow_function_parameter_dependency() {
    let file_path = "tests/typescript/fixtures/arrow_function_parameter_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "arrow_function_parameter_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "arrow_function_parameter_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}
