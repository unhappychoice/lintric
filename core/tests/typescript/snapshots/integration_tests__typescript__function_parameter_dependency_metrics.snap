---
source: core/tests/typescript/mod.rs
assertion_line: 60
expression: "serde_json::to_string_pretty(&result).unwrap()"
---
{
  "file_path": "tests/typescript/fixtures/function_parameter_dependency.ts",
  "line_metrics": [
    {
      "line_number": 2,
      "total_dependencies": 2,
      "dependency_distance_cost": 0.6666666666666666,
      "depth": 1,
      "transitive_dependencies": 1,
      "dependent_lines": [
        1,
        1
      ]
    }
  ],
  "overall_complexity_score": 3.266666666666667
}
